### PR TITLE
applications: nrf_desktop: Fix peer disconnection

### PR DESCRIPTION
**Pull request by MarekPieta**
Change fixes peer disconnection in ble_adv module. With previous implementation a central that was connected, but not yet bonded was not properly disconnected.

### CODE REVIEW

File: applications/nrf_desktop/src/modules/ble_adv.c [hunk 1]
```diff
@@ -172,6 +172,15 @@ static int ble_adv_stop(void)
 	return err;
 }
 
+static void conn_find(struct bt_conn *conn, void *data)
+{
+	struct bt_conn **temp_conn = data;
+
+	/* Peripheral can have only one Bluetooth connection. */
+	BUILD_ASSERT(CONFIG_BT_MAX_CONN == 1);
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/ble_adv.c:180]: check conn info and return conn only if indentity matches with the current one

Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_adv.c:180]: done

File: applications/nrf_desktop/src/modules/ble_bond.c [hunk 2]
```diff
@@ -693,6 +697,65 @@ static int init(void)
...
+{
+	if (state != STATE_ERASE_ADV) {
+		return false;
+	}
+
+	if (event->state == PEER_STATE_CONNECTED) {
+		/* Ensure that connected peer will have time to establish
+		 * security.
+		 */
+		if ((k_delayed_work_remaining_get(&timeout) < ERASE_ADV_NEW_CONN_TIMEOUT_MS) &&
```
Comment (joerchan) [inline applications/nrf_desktop/src/modules/ble_bond.c:710]:
```
 * Note that this behavior is
 * inherently subject to race conditions with the pre-existing
 * timeouts and work queue, so care must be taken to synchronize such
 * resubmissions externally.
```
Do we handle this properly? I don't know this module well enough to tell.

Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_bond.c:710]: This won't be a problem here. This work is submitted only from the context of the work queue thread.

**Pull request merged**